shotover/src/hot_reload/server.rs

```diff
@@ -184,15 +187,31 @@ pub fn start_hot_reload_server(socket_path: String, sources: &[Source]) {
         .collect();
 
     tokio::spawn(async move {
-        match UnixSocketServer::new(socket_path, source_handles) {
-            Ok(mut server) => {
-                info!("Unix socket server started for hot reload communication");
-                if let Err(e) = server.run().await {
-                    error!("Unix socket server error: {:?}", e);
+        const RETRY_DELAY: std::time::Duration = std::time::Duration::from_secs(3);
```

Comment:
why 3? why not 10 or 5? should this be another cli arg?
would we ever need to backoff?
should there be a bail counter? (ie if it fails 100 times give up and panic).